### PR TITLE
Make compiler read files from stdin

### PR DESCRIPTION
**Pull request by def-**
Special "-" file as stdin.

### CODE REVIEW

**Comment by Varriount** [comment]
Why not use a regular command line option?

**Comment by def-** [comment]
`-` is the convention for stdin/stdout in many programs. It's what I intuitively tried to check if this feature exists in Nim.

File: compiler/modules.nim [hunk 1]
```diff
@@ -116,7 +116,7 @@ proc newModule(fileIdx: int32): PSym =
   result.kind = skModule
   let filename = fileIdx.toFullPath
   result.name = getIdent(splitFile(filename).name)
-  if not isNimIdentifier(result.name.s):
+  if result.name.s != "-" and not isNimIdentifier(result.name.s):
```
Comment (Araq) [inline compiler/modules.nim]: If you fake the module name to be 'stdin', why do you need this check? Also the name 'stdin' will cause conflicts with 'system.stdin'.

Comment (def-) [inline compiler/modules.nim]: This is before the fake module name is set.

**Comment by def-** [comment]
Here's some of the fun things you can do with this:

```
$ nim -r c - <<< 'echo "Hello"; echo "World"'
Hello
World
$ nim -r c - <<EOL                                                                          ~
for i in 0..10:
  echo "Hello World"[0..i]
EOL
H
He
Hel
Hell
Hello
Hello 
Hello W
Hello Wo
Hello Wor
Hello Worl
Hello World
```
Also fun to combine with `--cc:tcc --verbosity:0`.

**Pull request merged**